### DISCUSSION TITLE
Specifiying which files to analyze to increase performance

### DISCUSSION THREAD

**Discussion by vzhan00** [spotbugs/spotbugs#3183] [Q&A]
I have a custom Spotbugs plugin implemented and was wondering if there was a way to specify which class files in the JAR to visit? There's only a small subset of class files that I actually want to analyze, and while I have code in the implemented BytecodeScanningDetector that doesn't scan a class if it doesn't match a regex, is there anything at a higher level that will prevent the plugin from even visiting the class to improve performance?

**Comment by iloveeclipse** [comment]
Simply repackage your jar and remove all classes you don't want to analyze?

**Reply by vzhan00** [reply]
This plugin is being vended to a lot of different users using it on different projects, so repackaging the JAR wouldn't be effective in my case.
